This patch prevents mutt from crashing when *buf is freed, the root cause is the
fact that an adjacent memory segment (*fmt) overruns and overwrite prev_size
field in the heap.

The bug and the patch were forwarded upstream, see http://bugs.mutt.org/3371

--- a/pager.c
+++ b/pager.c
@@ -1053,7 +1053,7 @@
     q = *fmt;
     while (*p)
     {
-      if (*p == '\010' && (p > *buf))
+      if (*p == '\010' && (p > *buf) && (q > *fmt))
       {
 	if (*(p+1) == '_')	/* underline */
 	  p += 2;
